### PR TITLE
Drtii 1339 healthcheck slack

### PR DESCRIPTION
**Pull request by ngu04**
Drtii 1339 : Adding slack notification for health check

### CODE REVIEW

File: src/main/scala/uk/gov/homeoffice/drt/Server.scala [hunk 1]
```diff
@@ -187,6 +190,20 @@ object Server {
...
+      val port = portCode.toString.toUpperCase
+      val link = urls.urlForPort(port)
+      val message =
+       s"""
+          Health Check Alert
+          port: $port
+          name: $checkName
+          priority: ${priority.toString}
+          link: $link
+        """
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/Server.scala]: Is there a reason for leaving blank lines, and starting every line with a 10 space indent?

File: src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.slf4j.LoggerFactory
+import uk.gov.homeoffice.drt.HttpClient
+
+import scala.concurrent.ExecutionContext
+
+trait SlackClient {
+  def notify(message: String)(implicit ec: ExecutionContext, mat: Materializer): Unit
+}
+
+case class SlackClientImpl(httpClient: HttpClient, webhookUrl: String) extends SlackClient {
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala:14]: How do we specify the channel messages are sent to?

Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala:14]: Can we add tests for this new client?

File: src/main/scala/uk/gov/homeoffice/drt/Server.scala [hunk 3]
```diff
@@ -196,10 +213,15 @@ object Server {
...
+    val soundAlarm = (portCode: PortCode, checkName: String, priority: IncidentPriority) => {
       sendEmail(portCode, checkName, priority, serverConfig.healthCheckTriggeredNotifyTemplateId)
-    val silenceAlarm = (portCode: PortCode, checkName: String, priority: IncidentPriority) =>
+      sendSlackNotification(portCode, checkName, priority)
+    }
+
+    val silenceAlarm = (portCode: PortCode, checkName: String, priority: IncidentPriority) => {
       sendEmail(portCode, checkName, priority, serverConfig.healthCheckResolvedNotifyTemplateId)
+      sendSlackNotification(portCode, checkName, priority)
+    }
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/Server.scala]: The messages don't say if the alarm is triggered or resolved

**Review by richbirch — CHANGES_REQUESTED** [review]
Nice one.
A few things to address

File: src/main/scala/uk/gov/homeoffice/drt/Server.scala (hunk 1 above)
Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/Server.scala]: Just had few message in dev channel and they look okay with 10 space . But I think it we will be fine without spaces as well .

File: src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala (hunk 2 above)
Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala:14]: There only one webhook url which is specific to channel for a env (it is retrieved from secret) . Preprod would have alerts-preprod and prod would have alerts-prod .

Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/notifications/SlackClient.scala:14]: sure

File: src/main/scala/uk/gov/homeoffice/drt/Server.scala (hunk 3 above)
Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/Server.scala]: That is good point .

**Review by richbirch — APPROVED** [review]
Nice one mate

**Pull request merged**